code/obj/item/device/pda2/base_os.dm

```diff
@@ -733,7 +800,20 @@
 							if (src.master)
 								src.master.explode()
 
-					src.master.display_alert(alert_beep)
+					if(senderName in src.blocked_numbers)
+						return
+
+					var/previewtext = (signal.data["tag"] == "preview_message")
+					if(previewtext && ON_COOLDOWN(src, "preview_cooldown", 15 SECONDS))
```

Comment:
The lack of feedback about preview cooldown made me think the feature is broken at first. At least throw a boutput here with a message about the cooldown or something like that.

Reply:
Done! Moved the check to the ringtone program and made it output a message via the PDA about why the button isn't working.